app/screens/post_options/components/reaction_bar/index.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {of as of$} from 'rxjs';
+import {catchError, switchMap} from 'rxjs/operators';
+
+import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@constants/database';
+import {safeParseJSON} from '@utils/helpers';
+
+import ReactionBar from './reaction_bar';
+
+import type {WithDatabaseArgs} from '@typings/database/database';
+import type SystemModel from '@typings/database/models/servers/system';
+
+const DEFAULT_EMOJIS = [
+    'thumbsup',
+    'smiley',
+    'white_check_mark',
+    'heart',
+    'eyes',
+    'raised_hands',
+];
+
+const mergeRecentWithDefault = (recentEmojis: string[]) => {
+    const filterUsed = DEFAULT_EMOJIS.filter((e) => !recentEmojis.includes(e));
+    return recentEmojis.concat(filterUsed).splice(0, 6);
+};
+
+const enhanced = withObservables([], ({database}: WithDatabaseArgs) => ({
+    recentEmojis: database.
+        get<SystemModel>(MM_TABLES.SERVER.SYSTEM).
+        findAndObserve(SYSTEM_IDENTIFIERS.RECENT_REACTIONS).
+        pipe(
+            switchMap((recent) => of$(mergeRecentWithDefault(safeParseJSON(recent.value) as string[]))),
```

Comment:
safeParseJSON may return a string, and not a list of string. If you think recent.value may not be properly formed, you will have to handle this in a different way (like using parseJSON and let it be the one that throws the exception catched by catchError).

Reply:
recent reactions will always be stored as an array of string.  but I will keep this in mind during testing.  Thanks :)

Reply:
Recent reactions will always be a array of strings. The problem is that the value for whatever reason is not a well formed json (if we assume it will be always well form why do we need the safe parse version?). In that case, safeParseJSON (if I remember correctly) will return the string, instead of whatever was codified in the string.